src/client/AM.PA.MacOs/PersonalAnalytics/Trackers/EmotionTracker/EmotionTracker.swift

```diff
@@ -0,0 +1,158 @@
+        emotionalState.arousal = questionnaire.arousal
+
+        do {
+            try self.context?.save()
+        } catch let error {
+            print("It was not possible to save the last emotional state.")
+            print("ERROR DETAILS: \(error)")
+        }
+
+        print("Emotional state saved:")
+        print(emotionalState)
+
+    }
+
+    // MARK: Notification scheduling
+    func scheduleNotification(minutesSinceNow: Int? = nil) {
+
+        let notification = NSUserNotification()
+        let notificationCenter = NSUserNotificationCenter.default
+        let timeIntervalSinceNow: Int = (minutesSinceNow ?? (UserDefaults.standard.value(forKey: "timeInterval") as! Int))
+
+        notification.title = "What are you feeling?"
+        notification.subtitle = "REMEMBER: push the button on your smartband!"
+        notification.soundName = NSUserNotificationDefaultSoundName
+        notification.deliveryDate = Date(timeIntervalSinceNow: TimeInterval(exactly: timeIntervalSinceNow)!)
+
+        notification.hasActionButton = true
+        notification.otherButtonTitle = "Dismiss"
+        notification.actionButtonTitle = "Postpone"
+        var actions = [NSUserNotificationAction]()
+        let action1 = NSUserNotificationAction(identifier: "5m", title: "5 minutes")
+        let action2 = NSUserNotificationAction(identifier: "30m", title: "30 minutes")
+        let action3 = NSUserNotificationAction(identifier: "1h", title: "1 hour")
+        actions.append(action1)
+        actions.append(action2)
+        actions.append(action3)
+        // WARNING, private API
+        notification.setValue(true, forKey: "_alwaysShowAlternateActionMenu")
+        notification.additionalActions = actions
+
+
+
+        print("Time to wait for next notification:", TimeInterval(exactly: timeIntervalSinceNow)!)
+
+        // More optional notification settings
+
+        // Identifier: DANGEROUS, the notification didn't work
+        // notification.identifier = "Emotion Pop-up notification"
+
+        // Informative text
+        // notification.informativeText = "And remember to push the button on your smartband!"
+
+        // Actual notification scheduling
+        notificationCenter.scheduleNotification(notification)
+    }
+
+    // MARK: Export collected data to csv
+    func exportToCsv(destinationPath: URL) {
+
+        // csv string to be saved
+        let csv = csv_string()
+
+        do {
+            try csv.write(to: destinationPath, atomically: true, encoding: String.Encoding.utf8)
+        } catch {
+            print("Something went wrong trying to save the file.")
+            // Possible problems: failed to write file – bad permissions, bad filename, missing permissions, or more likely it can't be converted to the encoding
+        }
+
+    }
+
+    // TODO: refactoring needed
+    func csv_string() -> String {
+
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "dd/MM/yyyy-HH:mm:ss"
```

Comment:
Is it possible to get the format from the OS settings?
Especially formats using `/` as delimiter can easily be misunderstood depending on location.
(Alternatively `yyyy-MM-dd HH:mm:ss` always works.)

Reply:
I temporarily adopted the latter solution you proposed, thanks for the advice.